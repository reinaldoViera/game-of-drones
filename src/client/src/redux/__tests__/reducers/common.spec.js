import commonReducer from '../../reducer/common'
const actions = {
    ADD_ENTITY: 'ADD_ENTITY',
    UPD_ENTITY: 'UPD_ENTITY',
    ADD_ENTITIES: 'ADD_ENTITIES',
    SERVER_LOADING: 'SERVER_LOADING',
    ERROR_SERVER_LOADING: 'ERROR_SERVER_LOADING',
}
const customReducer = commonReducer(undefined, actions, 'players', {
    players: {}
})
describe('Common reducer', () => {
    test('should give initial state', () => {
        const state = customReducer(undefined, {
            type: "UNHANDLED"
        });
        expect(state).toMatchSnapshot()
    });
    test('should add Entities', () => {
        const state = customReducer(undefined, {
            type: actions.ADD_ENTITIES,
            payload: {
                'p1Id': {
                    id: 'p1Id',
                    name: 'Player1'
                },
                'p2Id': {
                    id: 'p2Id',
                    name: 'Player2'
                }
            }
        });
        expect(state).toMatchSnapshot()
    });
    test('should add Entities to existing state', () => {
        const state = customReducer({
            error: false,
            loading: false,
            players: {
                'p1Id': {
                    id: 'p1Id',
                    name: 'Player1'
                },
                'p2Id': {
                    id: 'p2Id',
                    name: 'Player2'
                }
            }
        }, {
            type: actions.ADD_ENTITIES,
            payload: {
                'p3Id': {
                    id: 'p3Id',
                    name: 'Player3'
                },
                'p4Id': {
                    id: 'p4Id',
                    name: 'Player4'
                }
            }
        });
        expect(state).toMatchSnapshot()
    });

})