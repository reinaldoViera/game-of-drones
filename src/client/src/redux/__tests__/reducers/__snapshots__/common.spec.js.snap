// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Common reducer should add Entities 1`] = `
Object {
  "error": false,
  "loading": false,
  "players": Object {
    "p1Id": Object {
      "id": "p1Id",
      "name": "Player1",
    },
    "p2Id": Object {
      "id": "p2Id",
      "name": "Player2",
    },
  },
}
`;

exports[`Common reducer should add Entities to existing state 1`] = `
Object {
  "error": false,
  "loading": false,
  "players": Object {
    "p1Id": Object {
      "id": "p1Id",
      "name": "Player1",
    },
    "p2Id": Object {
      "id": "p2Id",
      "name": "Player2",
    },
    "p3Id": Object {
      "id": "p3Id",
      "name": "Player3",
    },
    "p4Id": Object {
      "id": "p4Id",
      "name": "Player4",
    },
  },
}
`;

exports[`Common reducer should give initial state 1`] = `
Object {
  "error": false,
  "loading": false,
  "players": Object {},
}
`;
